configure

```diff
@@ -26,8 +26,29 @@
 # make sure that is called using bash.
 
 # Get an absolute path to this script, since that determines the top-level directory.
-this_script_dir=`dirname $0`
-this_script_dir=`cd $this_script_dir > /dev/null && pwd`
+source_path="${BASH_SOURCE[0]:-$0}"
```

Comment:
Can we (do we want to) rely the shell running this script itself supporting BASH_SOURCE and/or other potential bash-isms?

Reply:
It is explicitly a bash script so I believe it is OK to use bash-isms. I usually write pure POSIX scripts because of the issues with MacOS compatibility (they use a very old GPLv2 version of BASH) and compatibility with other shells but didn't feel it was an issue here.

Reply:
This script is just a very thin wrapper of the script in `make/autoconf/configure`. That script is explicitly called with `bash` so using bash-isms there (and in the rest of the configure script and build system) is fine. In this wrapper however, I think it's better to stick with compatible shell as much as possible. The shebang in this file isn't really relevant as we can't make files executable. I suspect plenty of users are still running this as `sh configure` out of habit.

I would recommend moving the space check to `make/autoconf/configure` instead.